examples/features/customloadbalancer/client/customroundrobin/customroundrobin.go

```diff
@@ -0,0 +1,178 @@
+	"google.golang.org/grpc/internal/balancer/balanceraggregator"
+	"google.golang.org/grpc/serviceconfig"
+)
+
+func init() {
+	balancer.Register(customRoundRobinBuilder{})
+}
+
+const customRRName = "custom_round_robin"
+
+type customRRConfig struct {
+	serviceconfig.LoadBalancingConfig `json:"-"`
+
+	// ChildPolicy is the child policy of this balancer. This will be hardcoded
+	// to a graceful switch config which wraps a pick first with no shuffling
+	// enabled.
+	ChildPolicy serviceconfig.LoadBalancingConfig `json:"childPolicy"`
+
+	// ChooseSecond represents how often pick iterations choose the second
+	// SubConn in the list. Defaults to 3. If 0 never choose the second SubConn.
+	ChooseSecond uint32 `json:"chooseSecond,omitempty"`
+}
+
+type customRoundRobinBuilder struct{}
+
+func (customRoundRobinBuilder) ParseConfig(s json.RawMessage) (serviceconfig.LoadBalancingConfig, error) {
+	// Hardcode a pick first with no shuffling, since this is a petiole, and
+	// that is what petiole policies will interact with.
+	gspf, err := balanceraggregator.ParseConfig(json.RawMessage(balanceraggregator.PickFirstConfig))
+	if err != nil {
+		return nil, fmt.Errorf("error parsing hardcoded pick_first config: %v", err)
+	}
+	lbConfig := &customRRConfig{
+		ChooseSecond: 3,
+		ChildPolicy:  gspf,
+	}
+
+	if err := json.Unmarshal(s, lbConfig); err != nil {
+		return nil, fmt.Errorf("custom-round-robin: unable to unmarshal customRRConfig: %v", err)
+	}
+	return lbConfig, nil
+}
+
+func (customRoundRobinBuilder) Name() string {
+	return customRRName
+}
+
+func (customRoundRobinBuilder) Build(cc balancer.ClientConn, bOpts balancer.BuildOptions) balancer.Balancer {
+	crr := &customRoundRobin{
+		ClientConn: cc,
+		bOpts:      bOpts,
+	}
+	crr.balancerAggregator = balanceraggregator.Build(crr, bOpts)
+	return crr
+}
+
+var logger = grpclog.Component("example")
+
+type customRoundRobin struct {
+	// All state and operations on this balancer are either initialized at build
+	// time and read only after, or are only accessed as part of its
+	// balancer.Balancer API (UpdateState from children only comes in from
+	// balancer.Balancer calls as well, and children are called one at a time),
+	// in which calls are guaranteed to come synchronously. Thus, no extra
+	// synchronization is required in this balancer.
+	balancer.ClientConn
+	bOpts balancer.BuildOptions
+
+	// hardcoded child config, graceful switch that wraps a pick_first with no
+	// shuffling enabled.
+	balancerAggregator balancer.Balancer
+
+	cfg *customRRConfig
+
+	// InhibitPickerUpdates determines whether picker updates from the child
+	// forward to cc or not.
+	inhibitPickerUpdates bool
+}
+
+func (crr *customRoundRobin) UpdateClientConnState(state balancer.ClientConnState) error {
+	crrCfg, ok := state.BalancerConfig.(*customRRConfig)
+	if !ok {
+		return balancer.ErrBadResolverState
+	}
+	crr.cfg = crrCfg
+	return crr.balancerAggregator.UpdateClientConnState(balancer.ClientConnState{
```

Comment:
Comment?

```go
// A call to UpdateClientConnState should always produce a new Picker.  That
// is guaranteed to happen since the aggregator will always call UpdateChildState
// in its UpdateClientConnState.
```

Reply:
Sure, added.